package/src/utils/removeReactionFromLocalState.ts

```diff
@@ -27,8 +27,23 @@ export const removeReactionFromLocalState = <
     (r) => !(r.user_id === user?.id && r.type === reactionType),
   );
 
-  if (message.reaction_counts?.[reactionType] && message.reaction_counts?.[reactionType] > 0) {
-    message.reaction_counts[reactionType] = message.reaction_counts[reactionType] - 1;
+  if (message.reaction_groups?.[reactionType]) {
+    if (
+      message.reaction_groups[reactionType].count > 0 ||
+      message.reaction_groups[reactionType].sum_scores > 0
+    ) {
+      message.reaction_groups[reactionType].count = message.reaction_groups[reactionType].count - 1;
+      message.reaction_groups[reactionType].sum_scores =
+        message.reaction_groups[reactionType].sum_scores - 1;
```

Comment:
```suggestion
      message.reaction_groups[reactionType].count = Math.max(0, message.reaction_groups[reactionType].count - 1);
      message.reaction_groups[reactionType].sum_scores =
        Math.max(0, message.reaction_groups[reactionType].sum_scores - 1);
```

A guard for just in case.. if these fields mismatched with 0

Reply:
Done